docs/demultiplexing.md

```diff
@@ -0,0 +1,174 @@
+                    "inputs": [
+                        {"input": {"node_name": "request",
+                                   "data_item": "pipeline_input_name"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output_A",
+                         "alias": "output_A"},
+                        {"data_item": "output_B",
+                         "alias": "output_B"}
+                    ]
+                },
+                {
+                    "name": "Model_B_node",
+                    "model_name": "Model_B",
+                    "type": "DL model",
+                    "inputs": [
+                        {"input_A": {"node_name": "Model_A_node",
+                                     "data_item": "output_A"}},
+                        {"input_B": {"node_name": "Model_A_node",
+                                     "data_item": "output_B"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                },
+                {
+                    "name": "Model_C_node",
+                    "model_name": "Model_C",
+                    "type": "DL model",
+                    "gather_from_node": "Model_A_node",  <----------------- this parameter specifies which node to gather from (must be demultiplexer and in LIFO order)
+                    "inputs": [
+                        {"input": {"node_name": "Model_B_node",
+                                     "data_item": "output"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"pipeline_output_name": {"node_name": "Model_C_node",
+                                          "data_item": "output"}}
+            ]
+        }
+    ]
+}
+```
+
+## Pipeline configuration rules
+There are several rules for possible configurations in regards to demultiplexing and gathering:
+- You can gather only from nodes with `demultiply_count` specified (demultiplexer nodes)
+- When pipeline with dynamic `demultiply_count` encounters 0 results execution is stopped and gRPC/REST response returns specific error with such information
+- For pipelines with dynamic `demultiply_count` only 1 demultiplexer node is allowed.
+- Demultiplexer nodes and gathering nodes should be in LIFO order. Meaning you need to gather nodes starting from closest demultiplexer going upstream in Directed Acyclic Graph
+
+![diagram](demultiplexing_restriction_lifo.svg)
+
+- Node inputs must be coming from the same level of nested demultiplexing. To reduce demultiplexing levels you can gather results at any stage using `gather_from_node` parameter
+
+![diagram](demultiplexing_restriction_session_levels.svg)
```

Comment:
We should include also a section with current limitations like input BS and models BS

Reply:
Done